drivers/hmis/app/models/hmis/hud/processors/base.rb

```diff
@@ -123,4 +144,31 @@ def process_custom_field(field, value)
     record.assign_attributes(custom_data_elements_attributes: Array.wrap(cde_attributes))
     true
   end
+
+  # Get attributes for nested record(s)
+  def construct_nested_attributes(attribute_name, value, additional_attributes = {})
+    values = Array.wrap(value)
+
+    object_type = graphql_type(attribute_name) # eg the ClientName type
+
+    # Construct attribute objects for creating/updating records
+    attributes = values.map do |attribute_hash|
```

Comment:
I think this means the argument `value` would need to be either in the format:
```
[
  [a, b],
]
```
or
```
[
  { a: b },
]
```
or
```
{a: b}
```
is that right?
I wonder if we could enforce that a bit up on line 150.  
One thing to note is:
```
Array.wrap([:a, :b])
=> [:a, :b]
```
Which would mean if you passed in a simple array, I think this would break....

Reply:
Yeah, thats right. I would expect it to be a hash or an array of hashes. I added a line to raise an exception if any element is not a hash, on line 157 below.